tests/integration/test_iterators.py

```diff
@@ -0,0 +1,334 @@
+                metta_animal_base_handles.inheritance_human_mammal,
+                metta_animal_base_handles.inheritance_monkey_mammal,
+                metta_animal_base_handles.inheritance_chimp_mammal,
+                metta_animal_base_handles.inheritance_mammal_animal,
+                metta_animal_base_handles.inheritance_reptile_animal,
+                metta_animal_base_handles.inheritance_snake_reptile,
+                metta_animal_base_handles.inheritance_dinosaur_reptile,
+                metta_animal_base_handles.inheritance_triceratops_dinosaur,
+                metta_animal_base_handles.inheritance_earthworm_animal,
+                metta_animal_base_handles.inheritance_rhino_mammal,
+                metta_animal_base_handles.inheritance_vine_plant,
+                metta_animal_base_handles.inheritance_ent_plant,
+                metta_animal_base_handles.similarity_monkey_human,
+                metta_animal_base_handles.similarity_chimp_human,
+                metta_animal_base_handles.similarity_monkey_chimp,
+                metta_animal_base_handles.similarity_earthworm_snake,
+                metta_animal_base_handles.similarity_triceratops_rhino,
+                metta_animal_base_handles.similarity_vine_snake,
+                metta_animal_base_handles.similarity_ent_human,
+            ]
+        )
+
+    def _check_asserts(
+        self, das: DistributedAtomSpace, iterator: Union[LocalGetLinks, RemoteGetLinks]
+    ):
+        current_value = iterator.get()
+        assert current_value['handle'] == das.get_atom(iterator.get()['handle'])['handle']
+        assert isinstance(current_value, dict)
+        assert iterator.is_empty() is False
+        link_handles = sorted([item['handle'] for item in iterator])
+        assert len(link_handles) == 26
+        assert link_handles == self._expression_links()
+        assert iterator.is_empty() is True
+        with pytest.raises(StopIteration):
+            iterator.get()
+        with pytest.raises(StopIteration):
+            next(iterator)
+
+    def test_get_links_with_das_ram_only(self):
+        das = DistributedAtomSpace()
+        load_metta_animals_base(das)
+        iterator = das.get_links('Expression', no_iterator=False)
+        self._check_asserts(das, iterator)
+
+    def test_get_links_with_das_redis_mongo(self, _cleanup):
+        _db_up()
+        das = DistributedAtomSpace(
+            query_engine='local',
+            atomdb='redis_mongo',
+            mongo_port=mongo_port,
+            mongo_username='dbadmin',
+            mongo_password='dassecret',
+            redis_port=redis_port,
+            redis_cluster=False,
+            redis_ssl=False,
+        )
+        load_metta_animals_base(das)
+        das.commit_changes()
+        iterator = das.get_links('Expression', no_iterator=False, cursor=0)
+        self._check_asserts(das, iterator)
+        _db_down()
+
+    # Error with get_matched... in Adapter
```

Comment:
Do we have a card to uncomment this test?

Reply:
I'll change this comment. The real problem is that using load with MeTTa parse, the `template` table keys are not being saved as a string

Reply:
@marcocapozzoli 
OK. So we should start using this: https://docs.pytest.org/en/latest/how-to/skipping.html
Instead of commenting out tests.